python/desc/sims/GCRCatSimInterface/SedFitter.py

```diff
@@ -1,74 +1,104 @@
-import numpy as np
 import os
+import re
+import numpy as np
+import GCRCatalogs
+from lsst.utils import getPackageDir
+from lsst.sims.photUtils import BandpassDict, Bandpass, Sed, CosmologyObject
 
 __all__ = ["sed_from_galacticus_mags"]
 
+_gcr_sed_re = re.compile(r'sed_(\d+)_(\d+)_(bulge|disk)$')
+_galaxy_sed_dir = os.path.join(getPackageDir('sims_sed_library'), 'galaxySED')
 
-_LSST_IS_AVAILABLE = True
-try:
-    from lsst.utils import getPackageDir
-    from lsst.sims.photUtils import CosmologyObject
-except ImportError:
-    _LSST_IS_AVAILABLE = False
-
-
-def sed_from_galacticus_mags(galacticus_mags, redshift, h=0.71, omega_m=0.265):
+def _get_sed_mags_and_cosmology(catalog_name):
+    """
+    Returns 3 numpy arrays: sed_names, sed_mag_list, sed_mag_norm
+    and a dictionary for cosmology
+    sed_names is 1d str array, with length = number of SED files in the library
+    sed_mag_list is MxN float array, with M = number of SED files in the library,
+        and N = number of top hat filters in the catalog
+    sed_mag_norm is 1d float array, with length = number of SED files in the library
+    cosmology = {'H0': H0, 'Om0': Om0}
+    """
+    gc = GCRCatalogs.load_catalog(catalog_name, config_overwrite=dict(md5=False))
+    cosmo = dict(H0=gc.cosmology.H0.value, Om0=gc.cosmology.Om0)
+
+    bp_params_raw = {'disk': set(), 'bulge': set()}
+    for q in gc.list_all_quantities():
+        m = _gcr_sed_re.match(q)
+        if m:
+            wav0, width, tag = m.groups()
+            bp_params_raw[tag].add((int(wav0), int(width)))
+    assert bp_params_raw['disk'] == bp_params_raw['bulge'], 'SEDs for disk and bulge do not match'
+    assert bp_params_raw['disk'], 'No SED found'
+
+    bp_params_sorted = sorted(bp_params_raw['disk'], key=lambda p: p[0])
+    wav_min = bp_params_sorted[0][0] * 0.1
```

Comment:
Add a comment pointing out that this 0.1 is there because CatSim thinks in nm and the GCR thinks in Angstroms.  I just spend an embarrassingly long time trying to figure out why my own code requires that ;)